server/src/test/java/org/elasticsearch/health/HealthPeriodicLoggerTests.java

```diff
@@ -542,25 +747,24 @@ public void testMetricsMode() {
                 .put(HealthPeriodicLogger.ENABLED_SETTING.getKey(), true)
                 .build()
         );
-
-        HealthPeriodicLogger spyHealthPeriodicLogger = spy(testHealthPeriodicLogger);
-        spyHealthPeriodicLogger.isHealthNode = true;
-        List<HealthIndicatorResult> results = getTestIndicatorResultsWithRed();
-
-        doAnswer(invocation -> {
-            spyHealthPeriodicLogger.resultsListener.onResponse(results);
-            return null;
-        }).when(spyHealthPeriodicLogger).tryToLogHealth();
+        testHealthPeriodicLogger.clusterChanged(new ClusterChangedEvent("test", stateWithLocalHealthNode, ClusterState.EMPTY_STATE));
+        assertTrue("local node should be the health node", testHealthPeriodicLogger.isHealthNode());
 
         assertEquals(0, metrics.size());
 
         SchedulerEngine.Event event = new SchedulerEngine.Event(HealthPeriodicLogger.HEALTH_PERIODIC_LOGGER_JOB_NAME, 0, 0);
-        spyHealthPeriodicLogger.triggered(event);
+        testHealthPeriodicLogger.triggered(event);
 
         assertEquals(0, logs.size());
         assertEquals(4, metrics.size());
     }
 
+    private void verifyLoggerIsReadyToRun(HealthPeriodicLogger healthPeriodicLogger) {
+        assertTrue("local node should be the health node", healthPeriodicLogger.isHealthNode());
+        assertTrue("health logger should be enabled", healthPeriodicLogger.enabled());
+        assertEquals("health logger is started", Lifecycle.State.STARTED, healthPeriodicLogger.lifecycleState());
+    }
```

Comment:
There are quite some preconditions before the logger can `tryToLogHealth(...)`, to simplify the debugging of concurrency tests, I wanted to have the ensure the preconditions are correct.